src/qfusion.cpp

```diff
@@ -487,15 +507,64 @@ bitCapInt QFusion::ForceMReg(bitLenInt start, bitLenInt length, bitCapInt result
 
 void QFusion::INC(bitCapInt toAdd, bitLenInt start, bitLenInt length)
```

Comment:
There’s a lot of duplicated code and loops between INC and INCC (and likely elsewhere in this PR).  Can that be pulled out into functions to reduce redundancy and simplify?

Reply:
Done.